src/sentry/preprod/api/endpoints/project_preprod_check_for_updates.py

```diff
@@ -46,22 +55,125 @@ def get(self, request: Request, project) -> Response:
+                filter_kwargs["artifact_type"] = PreprodArtifact.ArtifactType.XCARCHIVE
+
+            return filter_kwargs
 
         try:
-            preprod_artifact = PreprodArtifact.objects.filter(
-                project=project, main_binary_identifier=main_binary_identifier
-            ).latest("date_added")
+            filter_kwargs = get_base_filters()
+            filter_kwargs.update(
+                {
+                    "main_binary_identifier": main_binary_identifier,
+                    "build_version": provided_version,
+                }
+            )
+
+            if provided_build_configuration:
+                filter_kwargs["build_configuration"] = provided_build_configuration
+
+            preprod_artifact = PreprodArtifact.objects.filter(**filter_kwargs).latest("date_added")
         except PreprodArtifact.DoesNotExist:
-            return Response({"error": "Not found"}, status=404)
+            logger.warning(
+                "No artifact found for binary identifier with version %s", provided_version
+            )
 
-        if preprod_artifact.build_version and preprod_artifact.build_number:
+        if preprod_artifact and preprod_artifact.build_version and preprod_artifact.build_number:
             current = InstallableBuildDetails(
+                id=str(preprod_artifact.id),
                 build_version=preprod_artifact.build_version,
                 build_number=preprod_artifact.build_number,
+                app_name=preprod_artifact.app_name,
+                download_url=get_download_url_for_artifact(preprod_artifact, request),
+                created_date=preprod_artifact.date_added.isoformat(),
             )
-        else:
-            current = None
-        return Response(CheckForUpdatesApiResponse(current=current).dict())
+
+        # Get the update object - find the highest version available
+        # Get all build versions for this app and platform
+        filter_kwargs = get_base_filters()
+        if preprod_artifact:
+            filter_kwargs["build_configuration"] = preprod_artifact.build_configuration
+        all_versions = (
+            PreprodArtifact.objects.filter(**filter_kwargs)
+            .values_list("build_version", flat=True)
+            .distinct()
+        )
+
+        # Find the highest semver version
+        highest_version = None
+        for version in all_versions:
+            if version:
+                try:
+                    parsed_version = parse_version(version)
+                    if highest_version is None or parsed_version > parse_version(highest_version):
+                        highest_version = version
+                except Exception:
+                    # Skip invalid version strings
+                    continue
+
+        # Get all artifacts for the highest version
+        if highest_version:
+            filter_kwargs = get_base_filters()
+            filter_kwargs["build_version"] = highest_version
```

Comment:
don't you still want to consider the `build_configuration` for this case? in which case, instead of creating a new instance of `filter_kwargs`, why don't you just reuse the `new_build_filter_kwargs` object you already made?